src/main/java/net/logstash/logback/stacktrace/StackElementFilter.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.logstash.logback.stacktrace;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * Component in charge of accepting or rejecting {@link StackTraceElement elements} when computing a stack trace hash
+ */
+public abstract class StackElementFilter {
```

Comment:
One question:

I see in your impl that you use [Pattern.find()](https://docs.oracle.com/javase/7/docs/api/java/util/regex/Matcher.html#find())

But in your README examples of patterns, it looks like you could use [Pattern.match()](https://docs.oracle.com/javase/7/docs/api/java/util/regex/Matcher.html#match()) instead (probably better in terms of performances):

```xml
<encoder class="net.logstash.logback.encoder.LogstashEncoder">
  <throwableConverter class="net.logstash.logback.stacktrace.ShortenedThrowableConverter">
    <maxDepthPerThrowable>30</maxDepthPerThrowable>
    <maxLength>2048</maxLength>
    <shortenedClassNameLength>20</shortenedClassNameLength>
    <exclude>sun\.reflect\..*\.invoke.*</exclude>
    <exclude>net\.sf\.cglib\.proxy\.MethodProxy\.invoke</exclude>
    <evaluator class="myorg.MyCustomEvaluator"/>
    <rootCauseFirst>true</rootCauseFirst>
  </throwableConverter>
</encoder>
```

I guess changing the impl from `find()` to `match()` may screw up some existing projects using your library, but maybe that could be wise to mention it in the doc and present some optimized patterns (using `^` and `$`):

```xml
<encoder class="net.logstash.logback.encoder.LogstashEncoder">
  <throwableConverter class="net.logstash.logback.stacktrace.ShortenedThrowableConverter">
    <maxDepthPerThrowable>30</maxDepthPerThrowable>
    <maxLength>2048</maxLength>
    <shortenedClassNameLength>20</shortenedClassNameLength>
    <exclude>^sun\.reflect\..*\.invoke.*</exclude>
    <exclude>^net\.sf\.cglib\.proxy\.MethodProxy\.invoke$</exclude>
    <evaluator class="myorg.MyCustomEvaluator"/>
    <rootCauseFirst>true</rootCauseFirst>
  </throwableConverter>
</encoder>
```